### PR TITLE
Aml typing

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: src/andromeda.ml [hunk 1]
```diff
@@ -69,35 +69,24 @@ let options = Arg.align [
...
-            Config.interactive_shell := false ;
-            add_file ~lim:!lim ~once:false true fn)]),
-     "<lim> <file> Process <file> up to the end of the statement at character\
-      <lim>, do not enter interactive mode");
   ]
 
 (** Interactive toplevel *)
-let toplevel cmp =
+let interactive_shell state =
   Format.printf "Andromeda %s@\n[Type #help for help.]@." Build.version ;
```
Comment (andrejbauer) [inline src/andromeda.ml]: Does `#help` still work?

File: src/parser/input.mli [hunk 2]
```diff
@@ -111,11 +111,28 @@ and match_op_case = pattern list * pattern option * comp
...
+type ml_ty = ml_ty' * Location.t
+and ml_ty' =
+  | ML_Arrow of ml_ty * ml_ty
+  | ML_Prod of ml_ty list
+  | ML_TyApply of Name.ty * ml_ty list
+  | ML_Handler of ml_ty * ml_ty
+  | ML_Judgment
+
+type constructor_decl =
+  | ML_GADT of Name.constructor * ml_ty list * ml_ty
```
Comment (andrejbauer) [inline src/parser/input.mli]: It may be better to switch back to the ADT syntax since we are not doing GADTs.

File: tests/data.m31.ref [hunk 3]
```diff
@@ -1,3 +1,2 @@
-Data constructor cow is declared.
-Data constructor bull is declared.
-[cow, bull]
+Types cattle declared.
```
Comment (andrejbauer) [inline tests/data.m31.ref]: *Type* cattle is declared.

**Comment by SkySkimmer** [comment]
Ready to be pulled.

**Pull request merged**